### DISCUSSION TITLE
Argument -n in force-level1-csv not working properly

### DISCUSSION THREAD

**Discussion by corneliussenf** [davidfrantz/force#77] [Q&A] [answered]
Hi all, me again... 

I just queried my metadata catalogue using following statement 'force-level1-csd -n -c 0,70 -s S2A,S2B  metadata level1 level1/l1_pool.txt T32TQT,T33TUN', expecting that no action will be taken (as specified by -n, right?). However, the download started, regardless of the argument -n being set. Am I doing something wrong or is this a bug I should report?

I use Force 3.6.3 on a Linux server.

Cornelius

**Comment by skoenig-eo** [comment]
Hi all,

I cannot provide anything productive in this matter, but I can say that I had the same experience yesterday with a slightly different setting: 
`force-csd -n /home/simon/force/catalogue /media/storage/bb/L1 /media/storage/bb/L1.pool /media/storage/bb/aux/aoi.shp` on a Microsoft Azure Ubuntu VM w/ FORCE 3.6.5.

Cheers
Simon

**Comment by ernstste** [comment, marked as answer]
Hi Cornelius and Simon,

This is most likely related to the way that you have installed FORCE.

Two ways to work around this:

- do not use -n as first argument, but after other arguments
- use the long option --no-act instead

I did not include this piece of information in the documentation, as the issue doesn't arise when you install FORCE in the 'usual' way. 
Please let me know if you're still running into the same issue using either of the two 'workarounds'.

Stefan

**Reply by corneliussenf** [reply]
Hi Stefan! Placing -n second order did the trick. Thanks!

**Reply by davidfrantz** [reply]
Hey @ernstste ,
what is the "unusual" way of installing?
Cheers,
David

**Reply by ernstste** [reply]
I guess I'll have to redirect that question to Cornelius and Simon.

In my case this only happened when I was directly running the script without installing it to the default location.
We talked about it briefly, not sure if you remember. I wasn't able to find out what caused the issue in a reasonable amount of time and just left it at that since the issue never appeared when FORCE was installed. 
If anyone has a clue what's happening, I'll be more than happy to hear from them!

**Reply by corneliussenf** [reply]
I can't really tell "how" FORCE was installed, as this was done my our system administrator. He followed your instructions on installation, though ;-)

**Reply by davidfrantz** [reply]
Ah okay, I remember vaguely..
I am more and more inclined to suggest Docker as the go-to solution to use FORCE ;)

**Reply by skoenig-eo** [reply]
Hmm 🤔 I chose the "regular" way according to the documentation, but I'll try the workaround and I hope that at some point I can switch to Docker.

Thanks for the helpful suggestions!

Best
Simon

**Reply by ernstste** [reply]
Okay thanks Cornelius and Simon. What is the output of `which force-level1-csd`

@davidfrantz Docker was working fine back then, but I haven't tested it again since you started providing images, David.

**Reply by corneliussenf** [reply]
I get: /usr/local/bin/force-level1-csd

**Reply by skoenig-eo** [reply]
Me too: `/usr/local/bin/force-level1-csd`

**Reply by skoenig-eo** [reply]
And the workaround with placing `-n` at the end did the trick. Thanks!

**Reply by ernstste** [reply]
Thank you!

I just tested it on our machines once more and the problem does seem to occur again. I'm not sure what has changed, but we'll look into it. Thanks for bringing my attention to the issue!

**Reply by maawoo** [reply]
I ran into this problem as well a while ago and kind of forgot to report it here because I "fixed" it somewhere along the line by using `--no-act`. 
My "unusual" way of installing is a local Singularity container that was build/converted from the official Docker image. 
I just ran `which force-level1-csd` as well on the two containers I currently have:

v3.6.3 container: `/usr/local/bin/force-level1-csd`
v3.6.5 container: `/home/docker/bin/force-level1-csd`

And I can confirm the described behavior for both containers.

**Reply by davidfrantz** [reply]
Alright, thanks all. 

I guess we have now confirmed that the installation does not matter. However, we also do not have any clue why this happens. We are using the ``getopt`` utility for this, and didn't find any obvious reason why this should not work. 
I will open an issue on this topic, and label it as ``help wanted`` - in case anybody experienced in bash wants to give it a look.
For the time being, I guess Stefan's answer is the way to go.

Cheers,
David

**Comment by ernstste** [comment]
We got to the bottom of the issue now. My apologies, this was indeed not related to the installation process, but rather a piece of code that probably was only included in my test setup but not our full installation at the time.

Thanks again @corneliussenf, @skoenig-eo, and @maawoo for bringing this up again. The fix will be deployed shortly.
